controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -468,10 +468,19 @@ private SegmentRange convert(final String scope,
 
     // Metrics reporting region
 
-    private void reportCreateStreamMetrics(String scope, String streamName, int initialSegments, CreateStreamStatus.Status status,
+    private void reportCreateStreamMetrics(String scope, String streamName, CreateStreamStatus.Status status,
                                            Duration latency) {
         if (status.equals(CreateStreamStatus.Status.SUCCESS)) {
-            streamMetrics.createStream(scope, streamName, initialSegments, latency);
+            Supplier<Number> activeSegmentsValueSupplier = () -> {
+                try {
+                    List<StreamSegmentRecord> activeSegments = streamStore.getActiveSegments(scope, streamName, null, executor).get();
```

Comment:
This is blocking on a remote call (`streamStore.getActiveSegments`) which can result in a long wait if the call is retried. This blocks your thread here!

Reply:
Agree with @shiveshr . This will quickly exhaust the controller thread pool and lead to serious side effects. Please find another way of getting this value.

Reply:
Yes that's also my concern in the first place as I've asked Andrei and Shivesh even before this PR.
Now it seems the question becomes how to optimize the getActiveSegments. There will be a separate PR for this, however beyond r0.5.